client/state/data-layer/wpcom/me/notification/settings/index.js

```diff
@@ -40,7 +31,6 @@ registerHandlers( 'state/data-layer/wpcom/me/notification/settings/index.js', {
 	[ NOTIFICATION_SETTINGS_REQUEST ]: [
 		dispatchRequest( {
 			fetch: requestNotificationSettings,
-			onSuccess: updateSettings,
```

Comment:
This is strange, why would we fetch notification settings if we ignore the response?

It seems that this entire `NOTIFICATION_SETTINGS_REQUEST` data layer handler and the associated `requestNotificationSettings` action creator is never used except in tests. Can be deleted.

Instead we use the `fetchSettings` thunk with the `NOTIFICATION_SETTINGS_FETCH`. It fetches from exactly the same `/me/notifications/settings` endpoint. And is used.

Reply:
Yeah I think there's a lot to clean up here with the notifications behavior. I might split that out to a separate pull request to keep this one reasonably-sized.